src/lib/components/healthselectorv2/HealthSelector.component.js

```diff
@@ -0,0 +1,233 @@
+type optionValue = {
+  icon?: React.Node,
+  label?: string,
+  shortLabel?: string,
+  value?: string,
+  disabled?: boolean,
+};
+
+type optionsConfiguration = { [optionType]: optionValue };
+
+type Props = {
+  optionsConfiguration?: optionsConfiguration,
+  label?: String,
+  onChange?: (value: string) => void,
+  defaultValue?: string,
+};
+
+const SingleValueWrapper = styled.div`
+  color: ${defaultTheme.white};
+  margin-left: ${defaultTheme.spacing.sp2}
+  margin-right: ${defaultTheme.spacing.sp2}
+  max-width: calc(100% - 8px);
+  position: absolute;
+  display: flex;
+  align-items: center;
+  height: 21px;
+  top: 0;
+`;
+
+const CSSCircle = styled.span`
+  background-color: ${(props) => defaultTheme.brand[props.variant]};
+  border-radius: 50%;
+  display: flex;
+  width: 16px;
+  height: 16px;
+  align-items: center;
+  justify-content: center;
+`;
+
+const ThreeCirclesWrapper = styled.span`
+  display: flex;
+  width: 34px;
+  justify-content: center;
+  ${CSSCircle}:first-child {
+    margin-right: -20%;
+  }
+  ${CSSCircle}:last-child {
+    margin-left: -20%;
+  }
+  height: 16px;
+`;
+
+const ValueIcon = styled.span`
+  margin: 0 ${defaultTheme.spacing.sp8};
+  height: 100%;
+  width: 100%;
+`;
+
+const OptionIcon = styled.span`
+  width: 35px;
+  margin-right: ${defaultTheme.spacing.sp4};
+  display: flex;
+  justify-content: center;
+  align-items: center;
+`;
+
+const OptionWrapper = styled.div`
+  display: flex;
+  align-items: center;
+  height: 100%;
+  justify-content: flex-start;
+  overflow: hidden;
+  line-height: 18px;
+`;
+
+const OptionLabel = styled.span`
+  display: flex;
+  align-items: bottom;
+  white-space: nowrap;
+  overflow: hidden;
+  text-overflow: ellipsis;
+  display: block;
+}
+`;
+
+const SelectStyle = styled(Select)`
+  .sc-select__control {
+    background-color: ${defaultTheme.defaultTheme.darkRebrand.buttonSecondary};
+
+    &.sc-select__control--is-focused {
+      border: 2px dashed ${defaultTheme.defaultTheme.darkRebrand.selectedActive};
+      background-color: ${defaultTheme.defaultTheme.darkRebrand
+        .buttonSecondary};
+    }
+    .sc-select__value-container {
+      overflow: visible;
+    }
+
+    &.sc-select__control--menu-is-open {
+      .sc-select__indicator {
+        transform: rotate(180deg);
+      }
+    }
+  }
+`;
+
+const icons = {
+  all: (
+    <ThreeCirclesWrapper>
+      <CSSCircle variant="statusHealthy" />
+      <CSSCircle variant="statusWarning" />
+      <CSSCircle variant="statusCritical" />
+    </ThreeCirclesWrapper>
+  ),
+  healthy: <Icon name={'Check-circle'} color={'statusHealthy'} size={'lg'} />,
+  warning: (
+    <Icon name={'Exclamation-circle'} color={'statusWarning'} size={'lg'} />
+  ),
+  critical: <Icon name={'Times-circle'} color={'statusCritical'} size={'lg'} />,
+  unknown: <Icon name={'Info'} color={'infoPrimary'} size={'lg'} />,
+};
+
+const optionsDefaultConfiguration: optionsConfiguration = {
```

Comment:
If an application want to hide one of those configurations, it will need to redo this whole mapping which may be a bit cumbersome. 
In order to improve dev experience we could refactor this to individual configurations that are exported so that it is easy to compose with them from one of the app